packages/flame_tiled/lib/src/rectangle_bin_packer.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+
+/// A simple solution to packing a series of rectangles into a larger bin.
+///
+/// Packing different rectangles in the smallest area possible is an
+/// NP-Complete problem. We are not going for perfection here, just something
+/// that works.
+///
+/// Note: Chrome on Android has a maximum texture size of 4096x4096. kIsWeb is
+/// used to select the smaller texture and might overflow. Consider using
+/// smaller textures for web targets, or, pack your own atlas.
+class RectangleBinPacker {
+  final double maxX;
+  final double maxY;
+
+  /// The bins of free space that we can search.
+  late final List<Rect> bins = [Rect.fromLTWH(0, 0, maxX, maxY)];
+
+  RectangleBinPacker({
+    this.maxX = kIsWeb ? 4096 : 8192,
+    this.maxY = kIsWeb ? 4096 : 8192,
+  });
+
+  /// Finds a free space for a rectangle of lengths [width] and [height] in
+  /// the atlas.
+  Rect pack(double width, double height) {
```

Comment:
Can you link to any page that explains the bin packing algorithm? (new to me)

Reply:
https://en.wikipedia.org/wiki/Rectangle_packing#Packing_different_rectangles_in_a_minimum-area_rectangle

https://www.david-colson.com/2020/03/10/exploring-rect-packing.html